File: CSharp/OpenTraceability/Models/Events/SensorMetaData.cs
using OpenTraceability.Interfaces;
using OpenTraceability.Utility.Attributes;
using System;
using System.Collections.Generic;

namespace OpenTraceability.Models.Events
{
    public class SensorMetaData
    {
        [OpenTraceabilityJson("time")]
        [OpenTraceability("@time")]
        public DateTimeOffset? TimeStamp { get; set; }

        [OpenTraceabilityJson("deviceID")]
        [OpenTraceability("@deviceID")]
        public Uri DeviceID { get; set; }

        [OpenTraceabilityJson("deviceMetadata")]
        [OpenTraceability("@deviceMetadata")]
        public Uri DeviceMetaData { get; set; }

        [OpenTraceabilityJson("rawData")]
        [OpenTraceability("@rawData")]
        public Uri RawData { get; set; }

        [OpenTraceabilityJson("startTime")]
        [OpenTraceability("@startTime")]
        public DateTimeOffset? StartTime { get; internal set; }

        [OpenTraceabilityJson("bizRules")]
        [OpenTraceability("@bizRules")]
        public Uri BizRules { get; internal set; }

        [OpenTraceabilityJson("dataProcessingMethod")]
        [OpenTraceability("@dataProcessingMethod")]
        public Uri DataProcessingMethod { get; internal set; }

        [OpenTraceabilityJson("endTime")]
        [OpenTraceability("@endTime")]
        public DateTimeOffset? EndTime { get; internal set; }

        [OpenTraceabilityExtensionAttributes]
        public List<IEventKDE> ExtensionAttributes { get; internal set; } = new List<IEventKDE>();
    }
}

File: JDK11/opentraceability/src/main/java/opentraceability/models/events/sensormetadata.java
package opentraceability.models.events;

import java.net.URI;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;

import opentraceability.interfaces.IEventKDE;
import opentraceability.utility.attributes.OpenTraceabilityAttribute;
import opentraceability.utility.attributes.OpenTraceabilityExtensionAttributesAttribute;
import opentraceability.utility.attributes.OpenTraceabilityJsonAttribute;

public class SensorMetaData {

    @OpenTraceabilityJsonAttribute(name="time")
    @OpenTraceabilityAttribute("", "@time")
    public OffsetDateTime timeStamp;

    @OpenTraceabilityJsonAttribute(name="deviceID")
    @OpenTraceabilityAttribute("", "@deviceID")
    public URI deviceID;

    @OpenTraceabilityJsonAttribute(name="deviceMetadata")
    @OpenTraceabilityAttribute("", "@deviceMetadata")
    public URI deviceMetaData;

    @OpenTraceabilityJsonAttribute(name="rawData")
    @OpenTraceabilityAttribute("", "@rawData")
    public URI rawData;

    @OpenTraceabilityJsonAttribute(name="startTime")
    @OpenTraceabilityAttribute("", "@startTime")
    public OffsetDateTime startTime;

    @OpenTraceabilityJsonAttribute(name="bizRules")
    @OpenTraceabilityAttribute("", "@bizRules")
    public URI bizRules;

    @OpenTraceabilityJsonAttribute(name="dataProcessingMethod")
    @OpenTraceabilityAttribute("", "@dataProcessingMethod")
    public URI dataProcessingMethod;

    @OpenTraceabilityJsonAttribute(name="endTime")
    @OpenTraceabilityAttribute("", "@endTime")
    public OffsetDateTime endTime;

    @OpenTraceabilityExtensionAttributesAttribute
    public List<IEventKDE> extensionAttributes = new ArrayList<>();
}